spec/request_handler/fieldsets_parser_spec.rb

```diff
@@ -33,41 +36,69 @@
+        let(:allowed) { {} }
+        let(:required) { [] }
+        let(:params) { {} }
+      end
     end
-  end
-  context 'fieldset settings and the parameter are set' do
-    it_behaves_like 'returns fieldsets' do
-      let(:params) { { 'fields' => { 'posts' => 'awesome' } } }
-      let(:expected) { { posts: [:awesome] } }
+    context 'fieldset settings and the parameter are set' do
+      it_behaves_like 'returns fieldsets' do
+        let(:params) { { 'fields' => { 'posts' => 'awesome' } } }
+        let(:expected) { { posts: [:awesome] } }
+      end
     end
-  end
 
-  context 'fieldset settings and multiple parameters are set' do
-    it_behaves_like 'returns fieldsets' do
-      let(:params)  { { 'fields' => { 'posts' => 'awesome,samples' } } }
-      let(:expected) { { posts: %i[awesome samples] } }
+    context 'fieldset settings and multiple parameters are set' do
+      it_behaves_like 'returns fieldsets' do
+        let(:params)  { { 'fields' => { 'posts' => 'awesome,samples' } } }
+        let(:expected) { { posts: %i[awesome samples] } }
+      end
     end
-  end
 
-  context 'fieldset settings and a required and an optional parameter are set' do
-    it_behaves_like 'returns fieldsets' do
-      let(:params) { { 'fields' => { 'posts' => 'awesome', 'photos' => 'foo' } } }
-      let(:expected) { { posts: [:awesome], photos: [:foo] } }
+    context 'fieldset settings and a required and an optional parameter are set' do
+      it_behaves_like 'returns fieldsets' do
+        let(:params) { { 'fields' => { 'posts' => 'awesome', 'photos' => 'foo' } } }
+        let(:expected) { { posts: [:awesome], photos: [:foo] } }
+      end
+    end
+
+    context 'fieldset settings and the required parameters are set' do
+      before do
+        opts.required = %i[posts photos]
+      end
+      it_behaves_like 'returns fieldsets' do
+        let(:params) { { 'fields' => { 'posts' => 'awesome', 'photos' => 'foo' } } }
+        let(:expected) { { posts: [:awesome], photos: [:foo] } }
+      end
     end
   end
 
-  context 'fieldset settings and the required parameters are set' do
-    before do
-      opts.required = %i[posts photos]
+  context 'fieldset types tests' do
+    context 'valid fieldset wich return all parameters because fieldset is set to true in RequestHandler config' do
+      it_behaves_like 'returns fieldsets' do
+        let(:params) { { 'fields' => { 'posts' => 'awesome', 'sample' => 'hello,moin,gutentach' } } }
+        let(:expected) { { posts: [:awesome], sample: %i[hello moin gutentach] } }
+      end
     end
-    it_behaves_like 'returns fieldsets' do
-      let(:params) { { 'fields' => { 'posts' => 'awesome', 'photos' => 'foo' } } }
-      let(:expected) { { posts: [:awesome], photos: [:foo] } }
+
+    context 'invalid fieldset wich fails because of unrecognized field in posts' do
+      it_behaves_like 'fails' do
+        let(:params) { { 'fields' => { 'posts' => 'awesome,good', 'sample' => 'hello,moin' } } }
```

Comment:
to make this test more clear please remove the fields options for the `sample` type